services/filemanager/digitalOceanSpacesManager.go

```diff
@@ -0,0 +1,197 @@
+package filemanager
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	SpacesManager "github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+// Upload passed in file to spaces
+func (manager *DOSpacesManager) Upload(file *os.File, prefixes ...string) (UploadOutput, error) {
+	if manager.Config.Bucket == "" {
+		return UploadOutput{}, errors.New("no storage bucket configured to uploader")
+	}
+
+	region := misc.GetSpacesLocation(manager.Config.EndPoint)
```

Comment:
return error if region is empty or does spaces have a default region and it works with empty string for that region